doc/develop/modules.rst

```diff
@@ -1021,6 +1021,36 @@ maps, each of which has the following entries:
 - ``doc-url``: A URL pointing to the location of the official documentation for
   this blob
 
+Package manager dependencies
+============================
+
+Zephyr modules can describe dependencies from external package managers,
+currently only ``pip`` is supported.
+
+A west extension command ``west packages <manager>`` is available to list
+package dependencies for Zephyr and the current modules.
+Run ``west help packages`` for more details.
+
+Python pip
+----------
+
+Calling ``west packages pip`` would list individual packages and requirement
+files. Passing ``--install`` would install these if there's an active virtual
```

Comment:
perhaps a link to [Requirements File Forma](https://pip.pypa.io/en/stable/reference/requirements-file-format/) to make it clear that requirements file refers to pip's feature and not a Zephyr specific thing.